src/cdk/tree/padding.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Input} from '@angular/core';
+import {CdkTreeNode} from './node';
+import {FlatNode} from './tree-data';
+
+/**
+ * Indent for the children tree nodes.
+ * This directive will add left-padding to the node to show hierarchy.
+ */
+@Directive({
+  selector: '[cdkNodePadding]',
+  host: {
+    '[style.padding-left]': 'paddingIndent',
+  }
+})
+export class CdkNodePadding<T extends FlatNode> {
+  /** The level of depth of the tree node. The padding will be `level * indent` pixels. */
+  @Input('cdkNodePadding') level: number;
+
+  /** The indent for each level. Default value is 28px. */
+  @Input('cdkNodePaddingIndex') indent: number = 28;
```

Comment:
This is marked as an `Input` but it has no setter. Also is `Index` the right term?